### ISSUE TITLE
ubuntu build failed with Source/CNTKv2LibraryDll/Trainer.cpp:526:57: error: format '%d' expects argument of type 'int', but argument 2 has type 'CNTK::DataUnit' 

### ISSUE THREAD

**Issue by lliimsft** [Microsoft/CNTK#2746]
Hi,

I am trying to build the latest 2.3.1, but failed with following error:

 Source/CNTKv2LibraryDll/Trainer.cpp: In member function 'size_t CNTK::Trainer::TotalNumberOfUnitsSeen(CNTK::DataUnit) const':
Source/CNTKv2LibraryDll/Trainer.cpp:526:57: error: format '%d' expects argument of type 'int', but argument 2 has type 'CNTK::DataUnit' [-Werror=format=]
             LogicError("Unsupported data unit: %d", unit);

Is there a int cast required in this line?

Thanks

**Comment by KeDengMS** [comment]
Yes, gcc5 has a more strict check, but our CI is still running gcc 4.8 so this got slipped by. Will fix soon.
